### PR TITLE
feat(ec2): extend BastionHostLinux to support CloudFormationInit

### LINKED ISSUE #17161: (aws-ec2): Extend BastionHostLinux to support CloudFormationInit
### Description

It would be helpful if the `BastionHostLinux` constructor accepted optional `CloudFormationInit` and  `ApplyCloudFormationInitOptions` arguments to be passed to the ​underlying `Instance`. This would enable full CloudFormation ​initialization of the instance like installing files.

```typescript
   ​/**
    ​* Apply the given CloudFormation Init configuration to the instance at startup.
    ​*
    ​* @default - no CloudFormation init
    ​* @stability stable
    ​*/
   ​readonly init?: CloudFormationInit;
   ​/**
    ​* Use the given options for applying CloudFormation Init.
    ​*
    ​* Describes the configsets to use and the timeout to wait
    ​*
    ​* @default - default options
    ​* @stability stable
    ​*/
   ​readonly initOptions?: ApplyCloudFormationInitOptions;
```

### Use Case

In my case I want to install files and run a command at initialization time.

### Proposed Solution

Add two optional arguments `init` and `initOptions` to `BastionHostLinuxProps`. In `BastionHostLinux` pass these arguments to the `Instance` constructor when creating the underlying instance.    

### Other information

_No response_

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by mpvosseller**
Implements https://github.com/aws/aws-cdk/issues/17161

Extends the `BastionHostLinux` constructor to accept optional `CloudFormationInit` and `ApplyCloudFormationInitOptions` arguments to be passed to the underlying instance.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ec2/lib/bastion-host.ts [hunk 1]
```diff
@@ -80,6 +81,22 @@ export interface BastionHostLinuxProps {
...
+   */
+  readonly init?: CloudFormationInit;
+
+  /**
+    * Use the given options for applying CloudFormation Init
+    *
+    * Describes the configsets to use and the timeout to wait
+    *
+    * @default - default options
+    */
```
Comment (njlynch) [inline packages/@aws-cdk/aws-ec2/lib/bastion-host.ts:98]: Super small nitpick; the spacing here is off (indented one space too many).

**Review by njlynch — CHANGES_REQUESTED** [review]
Looks good. Please add a test verifying the behavior

File: packages/@aws-cdk/aws-ec2/lib/bastion-host.ts (hunk 1 above)
Comment (mpvosseller) [reply packages/@aws-cdk/aws-ec2/lib/bastion-host.ts]: Fixed.

**Comment by mpvosseller** [comment]
Thank you @njlynch
I fixed the formatting and added a test.

**Comment by mpvosseller** [comment]
@njlynch the formatting is fixed, test is added, and rebased with master. Let me know if there is anything else I can do to get this merged. Thanks.

**Review by njlynch — APPROVED** [review]
Looks great! Apologies for the delay.

**Comment by njlynch** [comment]
FYI - For future reference, don't worry about updating from `master` (or other upstream branch) unless there are conflicts. Our automation will automatically update after approval and before merge. :)

**Comment by mpvosseller** [comment]
Thank you @njlynch!!!

**Comment by mpvosseller** [comment]
@njlynch What's the process to get this PR (already merged to `main`) merged into `v2-main`? Happy to open another PR if helpful but guessing it will automatically get merged in at some point? Thanks.

**Comment by njlynch** [comment]
We typically automatically merge forward daily; we put a temporary hold on the automation ahead of the v2 launch, but it's back on track now. This change is included in https://github.com/aws/aws-cdk/pull/17824, which will go out with the next v2 release.

**Pull request merged**